Core/Extensions/ExceptionMiddleware.cs

```diff
@@ -0,0 +1,81 @@
+﻿using FluentValidation;
+using FluentValidation.Results;
+using Microsoft.AspNetCore.Http;
+using System.Net;
+using WebNetSample.Core.CrossCuttingConcerns.Logging;
+using WebNetSample.Core.CrossCuttingConcerns.Logging.Log4Net;
+
+namespace WebNetSample.Core.Extensions;
+
+public class ExceptionMiddleware
+{
+    private RequestDelegate _next;
+    private LoggerServiceBase _loggerServiceBase;
+
+    public ExceptionMiddleware(RequestDelegate requestDelegate)
+    {
+        _next = requestDelegate;
+        _loggerServiceBase = new LoggerServiceBase("JsonFileLogger");
+    }
+
+    public async Task InvokeAsync(HttpContext httpContext)
+    {
+        try
+        {
+            await _next(httpContext);
```

Comment:
I couldn't get how it can handle in the same way like try-catch does. Should I put this code block below instead of `try-catch` blocks? 
```
var exceptionHandlerPathFeature =
               httpContext.Features.Get<IExceptionHandlerPathFeature>();
```

Reply:
There is an idea of [`UseExceptionHandler`](https://github.com/dotnet-fizzyy/Diploma/blob/V2/WebAPI/WebAPI/Startup/Configuration/ExceptionHandlerExtensions.cs#L15) method: it runs a separate task that process information about error response. Usage of [`httpContext.Features.Get<IExceptionHandlerFeature>()`](https://github.com/dotnet-fizzyy/Diploma/blob/V2/WebAPI/WebAPI/Startup/Configuration/ExceptionHandlerExtensions.cs#L22) will allow you to guarantee that exception has occured during request processing. It also provides other opportunities to check exception occurence during establishing connection to other services etc.

Reply:
I tried but it says `IApplicationBuilder` does not have `UseExceptionHandler`. Is it something you created or can it be from before the .net 6 version?

Also can you check my last [commit](https://github.com/osmanaliyardim/WebNetSample/pull/5/commits/d6d95f70b81caca12fb4deb4990b16595bd2003c)?

Reply:
Have you checked `WebApplication` class? It should have access probably

Reply:
I already have this in Program.cs as we talked in MS Teams